### PR TITLE
[Android] Fix CollectionView EmptyView

### LINKED ISSUE #10819: Android - Collectionview EmptyView not visible
### Description

Using a Collectionview with an EmptyView does not work.
The EmptyView is not being displayed. It does work on Windows, but not on Android.
If you set a HeightRequest on the CollectionView the EmptyView is being displayed.

### Steps to Reproduce

Just run the repo project and have a look.

### Link to public reproduction project repository

https://github.com/FM1973/EmptyViewProblem.git

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11+

### Did you find any workaround?

One can specify a HeightRequest and the EmptyView is beind displayed.
But this does prevent a dynamic height of the list.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix Android CollectionView EmptyView.

Captura de pantalla 2022-11-30 a las 14 51 19

### Issues Fixed

Fixes #10819

### CODE REVIEW

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs [hunk 1]
```diff
@@ -20,9 +18,36 @@ protected override void ConnectHandler(RecyclerView platformView)
...
+			var width = widthConstraint;
+			var height = heightConstraint;
+
+			if (!double.IsInfinity(width))
+				width = Context.ToPixels(width);
+
+			if (!double.IsInfinity(height))
+				height = Context.ToPixels(height);
+
+			UpdateEmptyViewSize(width, height);
```
Comment (Redth) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs:36]: Won't this try to force the RecyclerView to always grow to the maximum constraint size?  There's an issue on iOS where this already seems to happen and we want to make the behaviour more like Android to only grow to the size of its content if that does not exceed the constraints.

Comment (hartez) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs:36]:
```suggestion
            var result = base.GetDesiredSize(widthConstraint, heightConstraint);
			UpdateEmptyViewSize(result.Width, result.Height);
			return result;
```

Comment (hartez) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs:36]: Not a perfect solution, but it'll prevent the issue @redth is worried about.

**Review by hartez — CHANGES_REQUESTED** [review]
Questions about the empty view size changes.

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs [hunk 2]
```diff
@@ -21,9 +23,36 @@ protected override void ConnectHandler(RecyclerView platformView)
...
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs:47]: @jsuarezruiz the conversation was mark as resolved, but does not look like it was, can you please comment?

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs [hunk 3]
```diff
@@ -33,7 +33,9 @@ public override Size GetDesiredSize(double widthConstraint, double heightConstra
 			if (!double.IsInfinity(height))
 				height = Context.ToPixels(height);
 
-			UpdateEmptyViewSize(width, height);
+            var result = base.GetDesiredSize(widthConstraint, heightConstraint);
+			UpdateEmptyViewSize(result.Width, result.Height);
+			return result;
```
Comment (hartez) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs]: This seems suspect - the next line is unreachable.

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs [hunk 4]
```diff
@@ -31,6 +33,32 @@ protected override void DisconnectHandler(RecyclerView platformView)
...
+			var width = widthConstraint;
+			var height = heightConstraint;
+
+			if (!double.IsInfinity(width))
+				width = Context.ToPixels(width);
+
+			if (!double.IsInfinity(height))
+				height = Context.ToPixels(height);
+
+			UpdateEmptyViewSize(width, height);
```
Comment (hartez) [inline src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs:47]: If the ItemsView measurement is unconstrained in one direction, won't that set the height or width of the EmptyViewAdapter's RecyclerViewWidth or RecyclerViewHeight to infinity? It seems like it should use the height returned from GetDesiredSize, not the measurement constraints.

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs:47]: You are right, with the current approach the height is infinity.  However, with the return values from base.GetDesiredSize:

`DesiredSize {Width=0 Height=4793480}`

The EmptyView will not be visible using that values (zero Width).

**Review by hartez — APPROVED** [review]
Okay, looks like it's all working.

**Comment by janseris** [comment]
When is this fix released?

**Comment by hartez** [comment]
> When is this fix released?

At the moment, it's slated to be released as part of .NET 8. So it will likely be in the next .NET 8 preview.

**Comment by FreakyAli** [comment]
Wow, I am surprised that it took 2 years to fix something that is one of the major functionalities of CollectionView, I remember facing this issue with .Net 6 and it's coming out with .Net 8, Damn...

**Comment by mattleibow** [comment]
EmptyViewTemplate still does not render at all: https://github.com/dotnet/maui/issues/18551

**Pull request merged**